packages/server-wallet/src/wallet/store.ts

```diff
@@ -168,6 +203,36 @@ export class Store {
     return {outgoing, channelResult};
   }
 
+  async addMyState(
+    channel: Channel,
+    signedState: SignedStateWithHash,
+    tx: Transaction
+  ): Promise<Channel> {
+    const timer = timerFactory(this.timingMetrics, `addMyState ${channel.channelId}`);
```

Comment:
Should we be extra safe here and ensure that I am the one who actually signed this state? ie. something like
```
if (signedState.signature.signer !== channel.myAddress) throw 'Whoops!'
```

This would prevent a security issue where we accidentally add an untrusted state. (I hope that the name `addMyState` would be sufficient to prevent that, but I can imagine that `addMyState` eventually being renamed to something different.)

Reply:
Sounds very sensible. It's not entirely foolproof, as we could accidentally add a state that has `{signature: bogus, signer: me}`. But that seems much more unlikely. 

6b5d133